### PR TITLE
[Backup] `az backup restore restore-disks`: Add support for Cross Zonal Restore

### PR DESCRIPTION
**Pull request by akshayneema**
**Description**
- `az backup vault backup-properties set`: Added functionality to set vault storage redundancy to ZoneRedundant

- `az backup restore restore-disks`: Added support for Cross Zonal Restore(CZR) using --target-zone argument

- Some code clean-up

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/backup/_params.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def load_arguments(self, _):
         c.argument('tags', arg_type=tags_type)
 
     with self.argument_context('backup vault backup-properties set') as c:
-        c.argument('backup_storage_redundancy', arg_type=get_enum_type(['GeoRedundant', 'LocallyRedundant']), help='Sets backup storage properties for a Recovery Services vault.')
+        c.argument('backup_storage_redundancy', arg_type=get_enum_type(['GeoRedundant', 'LocallyRedundant', 'ZoneRedundant']), help='Sets backup storage properties for a Recovery Services vault.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/backup/_params.py:102]: Please use the active voice in the help description
```suggestion
        c.argument('backup_storage_redundancy', arg_type=get_enum_type(['GeoRedundant', 'LocallyRedundant', 'ZoneRedundant']), help='Set backup storage properties for a Recovery Services vault.')
```

File: src/azure-cli/azure/cli/command_modules/backup/_params.py [hunk 2]
```diff
@@ -349,6 +349,7 @@ def load_arguments(self, _):
         c.argument('disk_encryption_set_id', options_list=['--disk-encryption-set-id'], help='The disk encryption set id is used for encrypting restored disks. Please ensure access to disk encryption set id that is specified here.')
         c.argument('mi_system_assigned', action='store_true', help='Use this flag to specify whether a system-assigned managed identity should be used for the restore operation. MI option is not applicable for restoring unmanaged disks.')
         c.argument('mi_user_assigned', help='ARM ID of the user-assigned managed identity to use for the restore operation. Specify a value for this parameter if you do not want to use a system-assigned MI for restoring the backup item.')
+        c.argument('target_zone', arg_type=get_enum_type(['1', '2', '3']), help='Use this to specify the target zone number where the disks will be restored.')
```
Comment (wangzelin007) [inline src/azure-cli/azure/cli/command_modules/backup/_params.py:352]: Can you please add some explanation about the meaning of ['1', '2', '3'] ?

File: src/azure-cli/azure/cli/command_modules/backup/_validators.py [hunk 3]
```diff
@@ -42,3 +43,23 @@ def validate_crr(target_rg_id, rehydration_priority):
 
     if rehydration_priority is not None:
         raise MutuallyExclusiveArgumentError("Archive restore isn't supported for secondary region.")
+
+
+def validate_czr(backup_config_response, recovery_point):
+    backup_storage_redundancy = backup_config_response.properties.storage_type
```
Comment (wangzelin007) [inline src/azure-cli/azure/cli/command_modules/backup/_validators.py:49]: May I ask will backup_config_response.properties always have property storage_type or cross_region_restore_flag ?

Comment (akshayneema) [reply src/azure-cli/azure/cli/command_modules/backup/_validators.py:49]: Yes. These are compulsory parameters in the response. Attaching the response class for reference. 

image

File: src/azure-cli/azure/cli/command_modules/backup/_params.py (hunk 1 above)
Comment (akshayneema) [reply src/azure-cli/azure/cli/command_modules/backup/_params.py]: Done

File: src/azure-cli/azure/cli/command_modules/backup/_params.py (hunk 2 above)
Comment (akshayneema) [reply src/azure-cli/azure/cli/command_modules/backup/_params.py]: Done

**Review by zhoxing-ms — APPROVED** [review]

**Review by wangzelin007 — APPROVED** [review]

**Pull request merged**